cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -0,0 +1,166 @@
+package bicep
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/azure"
+	"github.com/azure/azure-dev/cli/azd/pkg/azureutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+
+	. "github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
+)
+
+func (p *BicepProvider) promptForParameter(ctx context.Context, key string, param azure.ArmTemplateParameter) (any, error) {
+	msg := fmt.Sprintf("Please enter a value for the '%s' infrastructure parameter:", key)
+	help, _ := param.Description()
+	azdMetadata, _ := param.AzdMetadata()
+	paramType := p.mapBicepTypeToInterfaceType(param.Type)
+
+	var value any
+
+	if paramType == ParameterTypeString && azdMetadata.Type != nil && *azdMetadata.Type == "location" {
```

Comment:
Do we want validation that prevents setting an `int` with `azdMetadata.Type` == `location`?

Reply:
We probably do - I can add a warning now, but maybe this is something an `azd lint` type thing will handle in the future?

Reply:
Actually - I am going to defer this for now, but I will add a note to the work tracking `azd lint`.